Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1

```diff
@@ -312,13 +312,27 @@ Describe "Tests for verifying Add-SslCert functionality" {
         }
     }
 
+    Context "Given hostnameport for cert does not exists" {
+
+        Mock Run-Command { return }
+        Mock Run-Command {return "", "" , "" , "", "", ""} -ParameterFilter { $failOnErr -eq $false }
+
+        $output = Add-SslCert -port "80" -certhash "SampleHash" -hostname "localhost" -sni "true" -iisVersion "8.0" 4>&1 | Out-String
+        
+        It "Should add hostnameport entry for given cert"{
+            ($output.Contains('netsh http show sslcert hostnameport=localhost:80')) | Should Be $true
+            Assert-MockCalled Run-command -Times 1 -Exactly -ParameterFilter { $failOnErr -eq $false }
+            Assert-MockCalled Run-command -Times 1
+        }
+    }
+
     Context "Given ipport for cert exists" {
 
         Mock Run-Command { return } -ParameterFilter { }
-        Mock Run-Command {return "", "" , "" , "", "IP:port                      : 0.0.0.0:80", "Certificate Hash             : samplehash"}  -ParameterFilter { $failOnErr -eq $false }
-
-        $output = Add-SslCert -port "80" -certhash "SampleHash" -hostname "localhost" -sni "true" -iisVersion "7.0" 4>&1 | Out-String
+        Mock Run-Command -Verifiable {return "", "" , "" , "", "IP:port                      : 0.0.0.0:80", "Certificate Hash             : samplehash"}  -ParameterFilter { $failOnErr -eq $false }
 
+        $output = Add-SslCert -ipAddress "0.0.0.0" -port "80" -certhash "SampleHash" -hostname "localhost" -sni "false" -iisVersion "8.0" 4>&1 | Out-String
```

Comment:
Update this test case for scenario where user passes ipAddress as "All Unassigned" from task UI